### PR TITLE
feat: support prompt setup in kotlin sdk

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

`prompt` value setup is now supported by our SDK.
* Change the refresh token in the `CodeTokenResponse` to be optional for we won't get a refresh token from it if the prompt value is `login`.(Refer to: https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1649906832335309)
* Change the error returned from `NO_REFRESH_TOKEN_FOUND` to `NOT_AUTHENTICATED` in the `getAccessToken` method when this method cannot retrieve a valid access token and try to refresh for a new one, but no refresh token is valid. (Refer to: https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1655869439813809)
* Fix Bug: `signOut` method in the `LogtoClient` should complete the process when no refresh token is provided.

## Testing

Test in the Android app.

### CODE REVIEW

**Review by IceHe — APPROVED** [review]
LGTM. Maybe better wait for another approval.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**
